src/genie/util/bitreader.impl.h

```diff
@@ -28,11 +40,40 @@ T BitReader::read(uint8_t s) {
 
 // ---------------------------------------------------------------------------------------------------------------------
 
+template <typename T, size_t SIZE, typename>
+T BitReader::readBypassBE() {
+    static_assert(SIZE > 0, "SIZE == 0");
+    static_assert(SIZE <= sizeof(T), "SIZE > sizeof(T)");
+    T ret = static_cast<T>(0);
+    istream.read(reinterpret_cast<char*>(&ret), SIZE);
+
+    // Swap Endianness if necessary
+    if (SIZE > 1) {
```

Comment:
Why do we need to swap the Endian for readBypassBE?

Reply:
x86 architecture is little endian, but data fields in MPEG-G are written out big endian. This is an implication required by part 2, 6.2:
"The bit order of syntax fields in the syntax tables is specified to start with the
MSB  and  proceed to the  LSB"-
The regular bitwriter does this automatically, when using bypass we need to change the order manually instead,